### PR TITLE
Add "impl" parameter to PickleType

### PR DESCRIPTION
**Pull request by yicheng47**
Add "impl" parameter to PickleType

### Description
1. add a impl parameter in PickleType's constructor to allow user to customize the underlying column type's implementation

### CODE REVIEW

File: lib/sqlalchemy/sql/sqltypes.py [hunk 1]
```diff
@@ -1798,11 +1802,20 @@ def __init__(
...
         """
         self.protocol = protocol
         self.pickler = pickler or pickle
         self.comparator = comparator
-        super(PickleType, self).__init__()
+
+        if impl:
+            self.impl = to_instance(impl)
+        else:
+            super(PickleType, self).__init__()
```
Comment (CaselIT) [inline lib/sqlalchemy/sql/sqltypes.py:1818]: I think it's best to always call super init, even in this case it would cause needles instantiation

**Review by CaselIT — COMMENTED** [review]
Looks fine, the test for this would be in tests/sql/test_types.py

File: lib/sqlalchemy/sql/sqltypes.py (hunk 1 above)
Comment (jason3gb) [reply lib/sqlalchemy/sql/sqltypes.py]: Fixed.

File: test/sql/test_types.py [hunk 2]
```diff
@@ -3809,6 +3809,11 @@ def test_nonmutable_comparison(self):
         ):
             assert p1.compare_values(p1.copy_value(obj), obj)
 
+    def test_customized_impl(self):
+        p1 = PickleType(impl=mysql.LONGBLOB)
+
+        assert isinstance(p1.impl, mysql.LONGBLOB)
```
Comment (CaselIT) [inline test/sql/test_types.py:3815]: let's also add an assert on a normal `PickleType` to verify it's still largebinary. something like `assert isinstance(PickleType().impl, LargeBinary)`

Comment (jason3gb) [reply test/sql/test_types.py:3817]: done

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision aabab0a97ba05f7463101fae4a974e51cb3e7479 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change aabab0a97ba05f7463101fae4a974e51cb3e7479: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

reckeck

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

openstack recheck

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889

**Comment by jason3gb** [comment]
@CaselIT Hi, I checked result of the last test, the error in the regression test seems not to be related with this current change. Could you help take another look at it? I'm a little confused here.

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

it probably just needs to rerun. sometimes some job can fail


recheck

recheck-openstack

openstack-recheck



View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889

**Comment by jason3gb** [comment]
@zzzeek formatting doc done, please take another look, thanks

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision e49bcd368d1f71dba92225d8d6e3af2bbe7142f7 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset e49bcd368d1f71dba92225d8d6e3af2bbe7142f7 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889

**Comment by CaselIT** [comment]
ugh, I already added a change to that gerrit. will re-apply it later

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision e49bcd368d1f71dba92225d8d6e3af2bbe7142f7 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset e49bcd368d1f71dba92225d8d6e3af2bbe7142f7 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2889 has been **merged**. Congratulations! :)

**Pull request closed without merging**